### PR TITLE
Add flexibility to reporters to choose output destination.

### PR DESCRIPTION
**Pull request by Patschkowski**
Enable setting alternate output destination for reporters instead of the default standard output sink.

### CODE REVIEW

File: include/aunit/framework/aunit-reporter.ads [hunk 1]
```diff
@@ -29,12 +29,17 @@
 --                                                                          --
 ------------------------------------------------------------------------------
 
+with Ada.Text_IO;
```
Comment (fedor-rybin) [inline include/aunit/framework/aunit-reporter.ads]: Ada.Text_IO is not available on most restricted runtimes like zfp for example. If you wich to take advantage of it you need to introduce different versions of affected packages for different runtime presets like is currently done for calendar, exceptions and memory management.

**Comment by Patschkowski** [comment]
Hi Fedor, what is the status of the review for this pull request? I hope my latest changes answer your remark. In case there is still something to do, please let me know.

File: include/aunit/reporters/aunit-reporter-xml.adb [hunk 2]
```diff
@@ -93,16 +94,16 @@ package body AUnit.Reporter.XML is
 
       Put_Line (File, "  <Statistics>");
       Put      (File, "    <Tests>");
-      Put (File, Integer (Test_Count (R)), 0);
+      Put (File, Trim (Test_Count (R)'Img, Left));
       Put_Line (File, "</Tests>");
       Put      (File, "    <FailuresTotal>");
-      Put (File, Integer (Failure_Count (R)) + Integer (Error_Count (R)), 0);
+      Put (File, Trim (Count_Type'Image (Failure_Count (R) + Error_Count (R)), Left));
```
Comment (fedor-rybin) [inline include/aunit/reporters/aunit-reporter-xml.adb]: This fails to build:

aunit-reporter-xml.adb:100:24: "Count_Type" is not visible
aunit-reporter-xml.adb:100:24: non-visible declaration at ada_containers.ads:20

File: include/aunit/framework/nocalendar/aunit-io.ads [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+--                                                                          --
+------------------------------------------------------------------------------
+
+with GNAT.IO;
+
+package AUnit.IO is
+
+   subtype File_Type is GNAT.IO.File_Type;
+
+   type File_Access is access constant GNAT.IO.File_Type;
```
Comment (fedor-rybin) [inline include/aunit/framework/nocalendar/aunit-io.ads]: Unfortunately that still won't work as in restricted runtimes there is no File_Type at all. GNAT.IO basically looks like this:

```
package GNAT.IO is
   pragma Preelaborate;

   procedure Put (X : Integer);
   --  Output integer to specified file, or to current output file, same
   --  output as if Ada.Text_IO.Integer_IO had been instantiated for Integer.

   procedure Put (C : Character);
   --  Output character to specified file, or to current output file

   procedure Put (S : String);
   --  Output string to specified file, or to current output file

   procedure Put_Line (S : String);
   --  Output string followed by new line to specified file, or to
   --  current output file.

   procedure New_Line (Spacing : Positive := 1);
   --  Output new line character to specified file, or to current output file

end GNAT.IO;```

Comment (Patschkowski) [reply include/aunit/framework/nocalendar/aunit-io.ads]: Ok, then I will implement File_Type/File_Access as dummy types for the restricted runtimes. Is there a way I can test this in advance?

Comment (fedor-rybin) [reply include/aunit/framework/nocalendar/aunit-io.ads]: That should work I think.
If you do not have acces to a cross toolchain, you can download gnat community edition for arm-elf target from AdaCore website, put it in path and then attempt to build aunit for any zfp runtime, for example like this:
```make TARGET=arm-eabi RTS=zfp RTS_CONF="--RTS=zfp-stm32f4"```

Comment (Patschkowski) [reply include/aunit/framework/nocalendar/aunit-io.ads]: Have a look at my latest commit, also checked this with the ARM cross compiler.

**Comment by fedor-rybin** [comment]
I appologize for the long wait. The latest version is indeed compatible with restricted runtimes.
Many thanks for your contribution and you patience.

**Pull request merged**